### DISCUSSION TITLE
[12.0 -> 13.0] Fail to delete constraint

### DISCUSSION THREAD

**Discussion by igallart** [OCA/OpenUpgrade#2786] [Q&A] [answered]
Hello, I'm a problem when openupgrade clean things, after migrate. Error:

2021-06-29 13:27:12,912 119522 INFO B1213 odoo.addons.base.models.ir_model: Deleting 6@stock.picking.type (stock_dropshipping.picking_type_dropship) 
2021-06-29 13:27:12,945 119522 ERROR B1213 odoo.sql_db: bad query: DELETE FROM stock_picking_type WHERE id IN (6)
ERROR: update or delete on table "stock_picking_type" violates foreign key constraint "stock_picking_picking_type_id_fkey" on table "stock_picking"
DETAIL:  Key (id)=(6) is still referenced from table "stock_picking".
 
2021-06-29 13:27:12,951 119522 WARNING B1213 odoo.modules.loading: Transient module states were reset 
2021-06-29 13:27:12,954 119522 ERROR B1213 odoo.modules.registry: Failed to load registry 
.......
.......

psycopg2.IntegrityError: update or delete on table "stock_picking_type" violates foreign key constraint "stock_picking_picking_type_id_fkey" on table "stock_picking"
DETAIL:  Key (id)=(6) is still referenced from table "stock_picking".

Any advice on how to fix it without deleting the operation in version 12?
Thanks.

**Comment by pedrobaeza** [comment]
Transferred as question, as it's not a problem of OpenUpgrade.

**Comment by pedrobaeza** [comment, marked as answer]
Are you using https://github.com/OCA/OpenUpgrade/pull/2379 ?

**Reply by igallart** [reply]
No, only openupgrade, I hadn't even seen that pull request..., Should I try what you say about the sql statement?

**Reply by pedrobaeza** [reply]
How do you expect that to work if no migration scripts are provided for such module? Check the current coverage.

**Reply by igallart** [reply]
Sorry, I didn't check because I mistakenly thought the V13 scripts were done. My mistake. Thanks.

**Comment by igallart** [comment]
@pedrobaeza , I did create the issue because I don't sure if it's not a problem of OpenUpgrade,@pedrobaeza , I created the issue because I don't know if it is a bug in openupgrade or in the migrated installation. The migrated installation has no problem apparently, everything works fine.
